src/main/java/org/spongepowered/common/item/inventory/custom/SpongeCustomInventoryBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.item.inventory.custom;
+
+import org.spongepowered.api.item.inventory.custom.CustomInventory;
+import org.spongepowered.api.item.inventory.custom.CustomInventory.Builder;
+import org.spongepowered.api.item.inventory.custom.InventoryType;
+import org.spongepowered.api.item.inventory.custom.InventoryTypes;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.common.text.SpongeTexts;
+
+import static org.spongepowered.api.item.inventory.custom.InventoryTypes.*;
+
+public class SpongeCustomInventoryBuilder implements CustomInventory.Builder {
+    private Text name;
+    private int size = 9 * 3;
+    private InventoryType type = InventoryTypes.CHEST;
+
+    @Override
+    public Builder name(Text name) {
+        this.name = name;
+        return this;
+    }
+
+    @Override
+    public Builder size(int size) {
+        this.size = size;
+        return this;
+    }
+
+    @Override
+    public CustomInventory build() {
+        // TODO no negative size
+        // TODO default name
+        return (CustomInventory) new SpongeCustomInventory(name, size, type);
+    }
+
+    @Override
+    public Builder type(InventoryType type) {
+        this.type = type;
+        if (type == CHEST) {
```

Comment:
Should we be changing the size here? In my opinion, we should let the size be set through the builder, and do verification when building the actual inventory, not here.